File: sources/com/s8/fwks/palm/topbar/TopbarIconTextButton.java
package com.s8.fwks.palm.topbar;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.fwks.palm.PalmWebSources;
import com.s8.pkgs.ui.carbide.icons.SVG_CarbideIcon;

/**
 * 
 */
public class TopbarIconTextButton extends S8WebObject {

	public static TopbarIconTextButton create(S8WebFront front, SVG_CarbideIcon icon, String text) {
		TopbarIconTextButton button = new TopbarIconTextButton(front);
		button.setIcon(icon);
		button.setText(text);
		return button;
	}
	
	
	/**
	 * 
	 */
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/topbar/TopbarIconTextButton";

	
	/**
	 * 
	 * @param front
	 */
	public TopbarIconTextButton(S8WebFront front) {
		super(front, WEBPATH);
	}

	
	/**
	 * 
	 * @param items
	 */
	public void setIcon(SVG_CarbideIcon icon) {
		vertex.outbound().setUInt16Field("icon", icon.code);
	}
	
	
	/**
	 * 
	 * @param items
	 */
	public void setText(String text) {
		vertex.outbound().setStringUTF8Field("text", text);
	}
	
	
}

File: web-sources/S8-fwks-palm/topbar/TopbarIconTextButton.js

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";
import { Popover } from "/S8-pkgs-ui-carbide/popover/Popover.js";

import { TopbarElement } from "./TopbarElement.js";


/**
 * 
 */
S8WebFront.CSS_import('/S8-fwks-palm/topbar/TopbarIconTextButton.css');


/**
 * 
 */
export class TopbarIconTextButton extends TopbarElement {


    /** @type{Popover} */
    tooltipPopover;

    /**
     * 
     */
    constructor() {
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("palm-topbar-ictxbutton");

        this.iconNode = document.createElement("div");
        this.iconNode.classList.add("palm-topbar-ictxbutton-icon");
        this.wrapperNode.appendChild(this.iconNode);

        this.textNode = document.createElement("div");
        this.textNode.classList.add("palm-topbar-ictxbutton-text");
        this.wrapperNode.appendChild(this.textNode);


        this.tooltipPopover = new Popover();
        this.tooltipPopover.setTheme("dark");
        this.tooltipPopover.setDirection("top");
        this.wrapperNode.appendChild(this.tooltipPopover.getEnvelope());

        this.tooltipContent = document.createElement("div");
        this.tooltipContent.classList.add("palm-topbar-ictxbutton-tooltip");
        this.tooltipContent.innerHTML = "This is a tooltip";
        this.tooltipPopover.node.appendChild(this.tooltipContent);

        const _this = this;
        this.tooltipPopover.hide();
        this.wrapperNode.addEventListener("mouseenter", function(){ 
            _this.tooltipPopover.show(); 
        }, false);
        this.wrapperNode.addEventListener("mouseleave", function(){ _this.tooltipPopover.hide(); }, false);
        this.wrapperNode.addEventListener("click", function(){ _this.tooltipPopover.hide(); }, false);
    }


    /**
     * 
     * @returns 
     */
    getEnvelope() {
        return this.wrapperNode;
    }



    /** continuous rendering approach */
    S8_render() { }

    /** continuous rendering approach */
    S8_dispose() { }

    /**
    * icon
    * @param {*} code 
    */
    S8_set_icon(code) {
        S8WebFront.SVG_insertByCode(this.iconNode, code, 16, 16);
    }

    /**
     * icon
     * @param {*} code 
     */
    S8_set_text(text) {
        this.textNode.innerHTML = text;
    }
}